train.py

```diff
@@ -1,216 +1,228 @@
+    # A Sqeuential model is a linear stack of layers. "Same" padding is used to
+    # ensure the output is the same size as the input. The input shape is the
+    # number of samples (input_size) and the number of features (1). The model
+    # is compiled with the Adam optimizer and the custom loss function "esr".
+    #
+    # The history object is used to store the results of model.fit and append
+    # metadata about the training process to the saved model file.
+    # =========================================================================
     clear_session()
+    history = None
     model = Sequential()
-    model.add(Conv1D(conv1d_filters, 12,strides=conv1d_strides, activation=None, padding='same',input_shape=(input_size,1)))
-    model.add(Conv1D(conv1d_filters, 12,strides=conv1d_strides, activation=None, padding='same'))
+    model.add(Input(shape=(input_size, 1)))
+    model.add(Conv1D(conv1d_filters, 12, strides=conv1d_strides, activation=None, padding='same'))
+    model.add(Conv1D(conv1d_filters, 12, strides=conv1d_strides, activation=None, padding='same'))
     model.add(LSTM(hidden_units))
     model.add(Dense(1, activation=None))
-    model.compile(optimizer=Adam(learning_rate=learning_rate), loss=error_to_signal, metrics=[error_to_signal])
-    print(model.summary())
-
-    # Load and Preprocess Data ###########################################
-    in_rate, in_data = wavfile.read(args.in_file)
-    out_rate, out_data = wavfile.read(args.out_file)
+    model.compile(optimizer=Adam(learning_rate=learning_rate), loss=Data.esr, metrics=[Data.esr])
+    print('\n') ;model.summary(); print('\n')
+
+    # =========================================================================
+    # Load and Preprocess Data
+    # -------------------------------------------------------------------------
+    # The input and output wav files are read and stored in numpy arrays. The
+    # data is normalized to the maximum value of the input data. The input data
+    # is then split into chunks of size "input_size" and the output data is
+    # shifted by "input_size-1" to align with the input data. The data is then
+    # shuffled and split into training and validation sets for training.
+    #
+    # NOTE: This currently uses only the first file in each data dir.
+    # Future versions likely to require multiple files for training.
+    # =========================================================================
+
+    in_sr, in_data = wavfile.read(in_files[0])
+    out_sr, out_data = wavfile.read(out_files[0])
+
+    # Validate that sample rates match
+    if(in_sr != out_sr):
+        # It's pretty trivial to do this conversion, but given that it's
+        # a strange situation to be in, preference on raising, as it
+        # probably indicates that something weird is going on.
+        raise ValueError("Input and output files have different sample rates.")
+    else:
+        sr = in_sr
```

Comment:
sample rates weren't being used much and there were hardcoded values, so this confirms that there's one known-good sample rate, and then uses that everywhere.